diff --git a/cpp/src/groupby/hash/compute_shared_memory_aggs.cu b/cpp/src/groupby/hash/compute_shared_memory_aggs.cu
--- a/cpp/src/groupby/hash/compute_shared_memory_aggs.cu
+++ b/cpp/src/groupby/hash/compute_shared_memory_aggs.cu
@@ -26,7 +26,7 @@
 namespace cudf::groupby::detail::hash {
 namespace {
 /// Shared memory data alignment
-CUDF_HOST_DEVICE cudf::size_type constexpr ALIGNMENT = 8;
+CUDF_HOST_DEVICE cudf::size_type constexpr ALIGNMENT = 16;
 
 // Allocates shared memory required for output columns. Exits if there is insufficient memory to
 // perform shared memory aggregation for the current output column.
